.github/workflows/build.yml

```diff
@@ -485,8 +487,12 @@ jobs:
         uses: dtolnay/rust-toolchain@nightly
         with:
           components: clippy
-      # do not deny warnings, as nightly clippy sometimes has false negatives
-      - run: ./admin/clippy
+      # Do not deny warnings, as nightly clippy sometimes has false negatives.
+      # - Ignore `clippy::disallowed_types` as a workaround since `disallowed_types` configured in `.clippy.toml`
+      #   is only intended for main `rustls` crate.
+      - run: ./admin/clippy -- --allow clippy::disallowed_types # --allow clippy::incompatible_msrv
+      # Check the main crate for all Clippy nightly warnings; do not deny.
+      - run: cargo clippy -p rustls
```

Comment:
Side question: these updates make me start to kinda wonder what is the point of keeping Clippy nightly as a CI job.

I think it may be ideal to keep this Clippy nightly CI job and deny only a subset of new Clippy nightly warnings, maybe just deny certain new Clippy nightly warnings as we see them suggest updating our code?

Reply:
I think they serve us well -- we will often notice them even as warnings only, which will avoid them breaking CI when new clippy lints are stabilized.

Reply:
Thanks `djc` that does make total sense to me. My apologies.

The one thing I want to say in my own defense is that I generally like to understand anything that I propose changing as much as I can, and like to clearly question in order to understand anything that doesn't make 100% sense to me. But yes I think I was (a little) out of line with the questioning and don't think my tone was right, my apologies.